### PR TITLE
LL-API and gym yamato tests

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
Add yamato tests that run sanity checks on the gym and low-level API interfaces. This code was lifted from the now-defunct notebooks (and cleaned up to make flake8 happy).

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
https://github.com/Unity-Technologies/ml-agents/tree/0.15.0/notebooks
https://jira.unity3d.com/browse/MLA-800
https://jira.unity3d.com/browse/MLA-816

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/tests/yamato/scripts/run_llapi.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import numpy as np
+
+from mlagents_envs.environment import UnityEnvironment
```
Comment (chriselion) [inline ml-agents/tests/yamato/scripts/run_llapi.py:3]: This and the gym code were copied from the old notebooks, and cleaned up slightly to remove matplotlib display, conform to flake8, and close the environment in a `finally`.

File: .yamato/gym-interface-test.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+test_editors:
+  - version: 2019.3
```
Comment (chriselion) [inline .yamato/gym-interface-test.yml:2]: I didn't think we needed to try these against multiple editor versions.

File: .yamato/gym-interface-test.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  name: Test Mac Gym Interface {{ editor.version }}
+  agent:
+    type: Unity::VM::osx
+    image: ml-agents/ml-agents-bokken-mac:0.1.4-492264
+    flavor: b1.small
+  variables:
+    UNITY_VERSION: {{ editor.version }}
+  commands:
+    - pip install pyyaml
+    - python -u -m ml-agents.tests.yamato.run_script_in_venv ml-agents/tests/yamato/scripts/run_gym.py
```
Comment (chriselion) [inline .yamato/gym-interface-test.yml:15]: I'm thinking about splitting this up into two commands - one to setup the venv (which could take options for additional python libraries) and another to just run the script on its own. Thoughts?

Comment (vincentpierre) [reply .yamato/gym-interface-test.yml:15]: Splitting would make it easier to read no ?

File: ml-agents/tests/yamato/run_script_in_venv.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    res = subprocess.run(
+        f"source {venv_path}/bin/activate; python {script_path}", shell=True
+    )
+
+    if res.returncode != 0:
+        print(f"{script_path} run FAILED!")
+        sys.exit(1)
+
+    print("{script_path} run SUCCEEDED!")
+    sys.exit(0)
```
Comment (vincentpierre) [inline ml-agents/tests/yamato/run_script_in_venv.py:26]:
```suggestion
    if res.returncode != 0:
        print(f"{script_path} run FAILED!")
        sys.exit(1)
    else:
        print("{script_path} run SUCCEEDED!")
        sys.exit(0)
```
Is this easier to read ?

File: ml-agents/tests/yamato/scripts/run_gym.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import numpy as np
+
+from gym_unity.envs import UnityEnv
+
+
+def main():
+
+    # Name of the Unity environment binary to launch
+    multi_env_name = "Project/testPlayer"
```
Comment (vincentpierre) [inline ml-agents/tests/yamato/scripts/run_gym.py:9]: Should this be a global variable to be used by both run_gym.py and run_llapi.py ?

File: .yamato/gym-interface-test.yml (hunk 3 above)
Comment (surfnerd) [reply .yamato/gym-interface-test.yml:15]: seems logical

File: ml-agents/tests/yamato/run_script_in_venv.py (hunk 4 above)
Comment (surfnerd) [reply ml-agents/tests/yamato/run_script_in_venv.py:26]: is the `else` needed in your suggested change? I was under the assumption that the program would exit after the call to `exit`

File: ml-agents/tests/yamato/scripts/run_gym.py (hunk 5 above)
Comment (surfnerd) [reply ml-agents/tests/yamato/scripts/run_gym.py:9]: I agree, either global variable or passing it to where it is need would be better than keeping it around multiple places.

Comment (chriselion) [reply ml-agents/tests/yamato/scripts/run_gym.py:9]: It should probably be a command line argument that defaults to this.

File: ml-agents/tests/yamato/run_script_in_venv.py (hunk 4 above)
Comment (chriselion) [reply ml-agents/tests/yamato/run_script_in_venv.py:26]: It's going to be a moot point when I move the venv setup to its own command

File: .yamato/gym-interface-test.yml (hunk 3 above)
Comment (chriselion) [reply .yamato/gym-interface-test.yml]: Done

File: ml-agents/tests/yamato/run_script_in_venv.py (hunk 4 above)
Comment (chriselion) [reply ml-agents/tests/yamato/run_script_in_venv.py]: Both scripts now take an `--env` commandline option (same syntax as `mlagents-learn`), which defaults to `Project/testPlayer`

**Pull request merged**